buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -681,6 +688,20 @@ class ClusterFormationTasks {
     static Task configureWaitTask(String name, Project project, List<NodeInfo> nodes, List<Task> startTasks, int waitSeconds) {
         Task wait = project.tasks.create(name: name, dependsOn: startTasks)
         wait.doLast {
+
+            nodes.forEach { node ->
+                Collection<String> unicastHosts = new HashSet<>()
+                nodes.forEach { otherNode ->
```

Comment:
Does this need to be nested? Could we calculate it once up-front and then write all the hosts files?

Reply:
Wouldn't we want to not have self-references in these (I guess it doesn't matter), but in that case we actually need additional code to account for the one self-reference that will return `null` in that loop.
I just figured this was the shortest solution? :)

Reply:
Ok, more seed node weirdness. I think you should up-front call `unicastTransportUri(node, null, project.ant)` in order to read the transport address of every node, and then put all of them into all of the `unicast_hosts.txt` files. It's normal for all the nodes to have the same list of unicast hosts, even though one of them refers to itself.